lib/IOExt/IOExt.h

```diff
@@ -1,33 +1,65 @@
-//
-// PCF8574 I/O Extension over I2C  !!! UNTESTED !!!
-//
+/*
+ * PCF8574 I/O Extension over I2C  !!! UNTESTED !!!
+ */
 
 #ifndef IOEXT_H
 #define IOEXT_H
 
 #include <PCF8574.h>
 #include <definitions.h>
+#include <abstract_task.h>
 
 class IOExt : public abstract_task {
 private:
-  // TODO: is there an IOExt1? If yes: we should integrate this here too, see ADC multi device as a reference, Else: remove '2'
-  PCF8574 IOExt2 = PCF8574(I2C_ADDRESS_PCF8574_IOExt2, I2C_SDA, I2C_SCL, I2C_INTERRUPT_PIN_PCF8574, keyPressedInterruptHandler);
+
+  PCF8574 IOExt[3] = {
+    PCF8574(I2C_ADDRESS_PCF8574_IOExt0, I2C_SDA, I2C_SCL, I2C_INTERRUPT_PIN_PCF8574, keyPressedInterruptHandler),
+    PCF8574(I2C_ADDRESS_PCF8574_IOExt1, I2C_SDA, I2C_SCL, I2C_INTERRUPT_PIN_PCF8574, keyPressedInterruptHandler),
+    PCF8574(I2C_ADDRESS_PCF8574_IOExt2, I2C_SDA, I2C_SCL, I2C_INTERRUPT_PIN_PCF8574, keyPressedInterruptHandler)
+  };
```

Comment:
,
PCF8574(I2C_ADDRESS_PCF8574_IOExt3, I2C_SDA, I2C_SCL, I2C_INTERRUPT_PIN_PCF8574, keyPressedInterruptHandler)

Reply:
we have 4 IOExt 
... and all resulting code too

Reply:
Thanks for the hint.